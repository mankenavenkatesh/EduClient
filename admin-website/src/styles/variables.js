export const black = "#000000";
export const white = "#ffffff";
export const brandBlue = "#0099cc";

export const brandDarkBlue = "#005f7f";
export const brandOrange = "#ff9933";
export const brandDarkOrange = "#ff6a33";
export const lightGrey = "#e5e5e5";
export const greyBlue = "#c1c9d1";
export const brandDark = "#324353";
export const faintBlue = "#f5f8fb";
export const faintGreen = "#f5fbf7";
export const faintOrange = "#fcf3e5";

export const headerUiBg = "#f3f8fc";
export const lowTranslucencyWhite = "rgba(255, 255, 255, 0.7)";

export const successColor = "#28a745";
export const warningColor = "#ffbf00";
export const warningBackground = "#fff9e8";
export const invalidColor = "#e46767";
export const invalidBackground = "#fbebe7";
