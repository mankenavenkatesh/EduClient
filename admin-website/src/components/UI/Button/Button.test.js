// Button.spec.js
import React from "react";
import jest from "jest-mock";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { matchers } from "jest-emotion";
import renderer from "react-test-renderer";
import { Button, OrangeButton, BlueButton } from ".";
import { brandOrange, brandBlue } from "../../../styles/variables";

expect.extend(matchers);

describe("Button component", () => {
  enzyme.configure({ adapter: new Adapter() });
  it("should be defined", () => {
    expect(Button).toBeDefined();
  });

  it("shows button text", () => {
    const button = renderer.create(<Button>Test Button</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it("shows pill button variant", () => {
    const button = renderer.create(<Button variant="pill">Test Button</Button>);
    const tree = button.toJSON();
    expect(tree).toHaveStyleRule("border-radius", "50px");
  });

  it("shows rounded button variant", () => {
    const button = renderer.create(
      <Button variant="rounded">Test Button</Button>
    );
    const tree = button.toJSON();
    expect(tree).toHaveStyleRule("border-radius", "5px");
  });

  it("shows orange button", () => {
    const button = renderer.create(<OrangeButton>Test Button</OrangeButton>);
    const tree = button.toJSON();
    expect(tree).toHaveStyleRule("background", brandOrange);
  });

  it("shows blue button", () => {
    const button = renderer.create(<BlueButton>Test Button</BlueButton>);
    const tree = button.toJSON();
    expect(tree).toHaveStyleRule("background", brandBlue);
  });

  it("should call mock function when button is clicked", () => {
    const mockFn = jest.fn();
    const tree = enzyme.shallow(<Button onClick={mockFn}>Test Button</Button>);
    tree.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
});
