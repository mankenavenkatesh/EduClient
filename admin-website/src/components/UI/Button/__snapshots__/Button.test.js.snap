// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component shows button text 1`] = `
<button
  className="css-1ocyg83-Button"
>
  Test Button
</button>
`;
